File: functions/alsa-lib/variants/go/0_alaw_to_s16_6.go
package main

func alawToS16(aVal byte) int32 {
	aVal ^= 0x55
	t := int32(aVal & 0x7f)
	if t < 16 {
		t = t<<4 + 8
	} else {
		seg := (t >> 4) & 0x07
		t = 0x108 + ((t & 0x0f) << 4)
		t = t << seg - 1
	}
	if aVal&0x80 != 0 {
		return t
	}
	return -t
}


func main(){
    alawToS16('a')
}

File: functions/alsa-lib/variants/c/0_alaw_to_s16_6.c
#include <stdint.h>

static int alaw_to_s16_v7(unsigned char a_val)
{
	int t = ((a_val ^ 0x55) & 0x7F) + 8;
	int seg = (t & 0x70) >> 4;
	t = (seg == 0 ? t << 4 : t << (seg + 3)) & 0x7FFF;
	return (a_val & 0x80) ? t : -t;
}


int main(){
    alaw_to_s16_v7('a');
}